src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -90,6 +97,10 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 		}
 	}
 
+	if err := r.updateStsImage(ctx, &sts); err != nil {
```

Comment:
I'd think that any change restarting the cluster containers should go through a partitioned rolling update (to avoid availability issues). We might need to abstract the upgrade logic here (and pass functions) to cater for other kinds of updates.

Reply:
yeah, since that's a bug in my code let's merge this and adjust the resource update to do that. Opened #656 thanks